### PR TITLE
Video welcome dialog (behind flag)

### PR DESCRIPTION
**Pull request by microbit-robert**
- Adds new dialog for live (video is a placeholder) behind livePreview flag.
- Same flag also disabled the black release notice bar (useful for screenshots etc)
- Add Feedback button to Help menu

### CODE REVIEW

File: src/common/YouTubeVideo.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+          <AspectRatio ratio={16 / 9}>
+            <iframe
+              // Avoid youtube cookie. rel=0 should limit related videos to our channel.
+              // Once we have translated videos we can try e.g. cc_lang_pref=fr
+              // but we'll need to check our codes match theirs.
+              title="welcome video"
+              src={`https://www.youtube-nocookie.com/embed/${
+                youtubeId ? youtubeId.trim() : ""
+              }?rel=0&cc_load_policy=1`}
+              allow="encrypted-media"
```
Comment (microbit-matt-hillsdon) [inline src/common/YouTubeVideo.tsx:30]: Does this work via the versioner at the moment?

Comment (microbit-robert) [reply src/common/YouTubeVideo.tsx:30]: I ran the versioner locally pointing to this review branch to test. The video played and there were no errors or warnings in the console.

File: src/workbench/InfoDialog.tsx [hunk 2]
```diff
@@ -4,33 +4,64 @@
...
+const InfoDialog = ({ isOpen, onClose }: InfoDialogProps) => {
   const { guideLink } = useDeployment();
+  const [welcomeVideo, setWelcomeVideo] = useState<YoutubeVideo | undefined>();
+  const [loadError, setLoadError] = useState<boolean>(false);
+  const logging = useLogging();
+  useEffect(() => {
+    const controller = new AbortController();
+    const { signal } = controller;
+    const fetchWelcomeVideo = async () => {
+      const query = `*[_type == "pythonEditorConfig"]{
```
Comment (microbit-matt-hillsdon) [inline src/workbench/InfoDialog.tsx:40]: Maybe query by _id? Then it's clearer that we only expect one. _id is also "pythonEditorConfig".

Comment (microbit-robert) [reply src/workbench/InfoDialog.tsx]: Makes sense. [`0d14718`](https://github.com/microbit-foundation/python-editor-next/pull/794/commits/0d14718730921fd631e434da706b8d7c5a5c10d9).

File: src/workbench/InfoDialog.tsx [hunk 3]
```diff
@@ -4,33 +4,64 @@
...
+        welcomeVideo
+       }`;
+      try {
+        const response = await fetch(queryUrl(query), { signal });
+        if (response.ok) {
+          const { result } = await response.json();
+          if (!result) {
+            throw new Error("Unexpected response format");
+          }
```
[the remark this answers is not in the document]
Comment (microbit-robert) [reply src/workbench/InfoDialog.tsx]: This now uses existing functions. I think we're in a position to use the `languageId` if we wanted to translate the `alt` [`a8604af`](https://github.com/microbit-foundation/python-editor-next/pull/794/commits/a8604af7397944d323e05a326467d359f51cce2b).

File: src/workbench/InfoDialog.tsx [hunk 4]
```diff
@@ -4,33 +4,63 @@
...
+        return result[0].welcomeVideo;
+      }
+    };
+    const query = (languageId: string): string => {
+      if (!languageId.match(/^[a-z-]+$/g)) {
+        throw new Error("Invalid language id.");
+      }
+      // Ready to use languageId here if the alt field is made translatable.
+      return `
+        *[_id == "pythonEditorConfig"]{
```
Comment (microbit-matt-hillsdon) [inline src/workbench/InfoDialog.tsx:50]: We should exclude drafts here (see other query).

Comment (microbit-robert) [reply src/workbench/InfoDialog.tsx]: [`56d8b38`](https://github.com/microbit-foundation/python-editor-next/pull/794/commits/56d8b38005280e77d983778e19a0083697db9312).

**Comment by microbit-matt-hillsdon** [comment]
@microbit-robert if we're not getting a video for a while we need to at least merge the strings from this for translation.

Tempted to either produce a very quick video or put the feature behind a flag so we can merge quickly. Let's discuss next week.

**Comment by microbit-matt-hillsdon** [comment]
Discussed with Lucy and agreed we'll have a stop-gap video for Friday so no need for a flag.

File: src/flags.ts [hunk 5]
```diff
@@ -52,5 +67,7 @@ export const flagsForParams = (stage: string, params: URLSearchParams) => {
 
 export const flags: Flags = (() => {
   const params = new URLSearchParams(window.location.search);
-  return flagsForParams(stageFromEnvironment, params);
+  const flags = flagsForParams(stageFromEnvironment, params);
+  console.log(flags);
+  return flags;
```
Comment (microbit-matt-hillsdon) [inline src/flags.ts:72]:
```suggestion
  return flagsForParams(stageFromEnvironment, params);
```

**Review by microbit-matt-hillsdon — APPROVED** [review]
Looking good now. @microbit-robert reviewed my recent changes.

**Pull request merged**